jsonschema_lexer/_lexer.py

```diff
@@ -24,109 +27,162 @@ class JSONSchemaLexer(JsonLexer):
     ]
 
     data_types: ClassVar[list[str]] = [
-        "object",
-        "integer",
-        "string",
-        "number",
-        "array",
-        "boolean",
-        "null",
-    ]
-    core_keywords: ClassVar[list[str]] = [
-        "$schema",
-        "$id",
-        "$ref",
-        "$defs",
-        "$comment",
-        "$dynamicAnchor",
-        "$dynamicRef",
-        "$anchor",
-        "$vocabulary",
-    ]
-    applicator_keywords: ClassVar[list[str]] = [
-        "oneOf",
-        "allOf",
-        "anyOf",
-        "if",
-        "then",
-        "else",
-        "not",
-        "properties",
-        "patternProperties",
-        "additionalProperties",
-        "dependentSchemas",
-        "propertyNames",
-        "prefixItems",
-        "contains",
-        "items",
-    ]
-    meta_data_keywords: ClassVar[list[str]] = [
-        "title",
-        "description",
-        "default",
-        "deprecated",
-        "examples",
-        "readOnly",
-        "writeOnly",
-    ]
-    validation_keywords: ClassVar[list[str]] = [
-        "type",
-        "enum",
-        "const",
-        "minLength",
-        "maxLength",
-        "pattern",
-        "maximum",
-        "exclusiveMinimum",
-        "multipleOf",
-        "exclusiveMaximum",
-        "minimum",
-        "dependentRequired",
-        "minProperties",
-        "maxProperties",
-        "required",
-        "minItems",
-        "maxItems",
-        "minContains",
-        "maxContains",
-        "uniqueItems",
-    ]
-    other_keywords: ClassVar[list[str]] = [
-        "format",
-        "unevaluatedItems",
-        "unevaluatedProperties",
-        "contentEncoding",
-        "contentMediaType",
-        "contentSchema",
-        "format_assertion",
+        '"object"',
+        '"integer"',
+        '"string"',
+        '"number"',
+        '"array"',
+        '"boolean"',
+        '"null"',
     ]
+    keywords: ClassVar[dict[str | None, list[str]]] = {}
```

Comment:
The key can be `None`? How/why?

Reply:
The line `self.identifier.get(dialect, None)` was giving an error stating: `No overloads for "get" match the provided arguments` as here dialect could also be None in which case this too should return None.

Reply:
Oh right, yes, `typing` is pretty terrible. OK.